src/board.rs

```diff
@@ -72,38 +76,35 @@ impl Board {
         }
     }
 
-    fn is_move_valid(&self, m: &Move) -> ChessResult<()> {
-        Self::is_in_bounds(&m.from)?;
-        Self::is_in_bounds(&m.to)?;
-        self.is_piece_some(&m.from)?;
-        if self.moves.contains(m) {
+    fn is_move_valid(&self, mv: &Move) -> ChessResult<()> {
+        Self::is_in_bounds(&mv.from)?;
+        Self::is_in_bounds(&mv.to)?;
+        self.is_piece_some(&mv.from)?;
+        if self.moves.contains(mv) {
             Ok(())
         } else {
             Err(ChessError::InvalidMove)
         }
     }
 
-    pub fn move_piece(&mut self, m: &Move) -> ChessResult<()> {
-        self.is_move_valid(&m)?;
-        let mut piece = self.get_piece(&m.from);
+    pub fn move_piece(&mut self, mv: &Move) -> ChessResult<()> {
+        self.is_move_valid(mv)?;
 
-        if let Some(Piece::Pawn(player)) = piece {
-            if (player == Player::White && m.to.y == 7) || (player == Player::Black && m.to.y == 0)
-            {
-                // TODO: always promote to queen for now, need to handle this eventually
-                piece = Some(Piece::Queen(player));
+        if let Some(mut piece) = self.get_piece_mut(&mv.from).take() {
+            if let Piece::Pawn(player) = piece {
+                if (player == Player::White && mv.to.y == BOARD_SIZE - 1)
+                    || (player == Player::Black && mv.to.y == 0)
```

Comment:
Can we use that code golf technique here as well?

Reply:
I almost kind of don't want that code golf now lol